src/legacy/core_plugins/kibana/public/discover/controllers/discover.js

```diff
@@ -561,11 +561,13 @@ function discoverController(
         });
 
         // update data source when filters update
-        $scope.$listen(queryFilter, 'update', function () {
-          $scope.filters = queryFilter.getFilters();
-          return $scope.updateDataSource().then(function () {
-            $state.save();
-          });
+        queryFilter.getUpdates$().subscribe({
```

Comment:
I'm not much into Angular, but I would expect there would be an `.unsubscribe()` in the treardown of `$scope`. But if the old code does not unsubscribe, I guess, we can leave that, too.

Reply:
Actually, the `.$listen` does "magically" unsubscribe

![image](https://user-images.githubusercontent.com/9773803/58023884-3841af00-7b11-11e9-9692-0afd48ff3cab.png)

so we need to `.unsubscribe()`, too.

Reply:
yep. done.